### ISSUE TITLE
Document Translation

### ISSUE THREAD

**Issue by skyao** [istio/istio.github.io#675]
hi, I'm Sky, I come from China. I'm interesting in istio and has setup a technical community in China for istio. We are trying to translate all these docs into Chinese,  almost 30% is done now.

See here:  https://doczhcn.gitbooks.io/istio/

Do you have any plan for document translation? We would like to contribute these Chinese documentation when we finish it.

**Comment by linsun** [comment]
Thanks for reaching out, we would be gladly accept PR for translation.

**Comment by skyao** [comment]
Where should we pull the Chinese content? We should not override the English content, we need some structure to put they, and I think you should also setup the auto build to publish the Chinese content.

Please help to tell us how and where we should put these translated content, we will follow and pull them.

**Comment by rshriram** [comment]
We don't have auto build. Its just pure github build. 
I think one good option is to tweak the webpage to have language selectors (en|mandarain etc..). When user chooses the content, you drop into the appropriate folder. This would mean that at top level you have a mandarain folder. and under the folder, you have the translated content (docs, blogs, etc.).

**Comment by geeknoid** [comment]
A concern I have is that translations will invariably be lagging relative to the current state of reality. We've moving to have more frequent releases soon. And we'll move the source code for istio.io into the main source repository so that it's easier to update docs when pushing code changes. As a result, it's likely that there will be an increasing rate of change in docs as the product evolves.

So putting the whole site into a particular language at a top level might mean the user is seeing old or missing docs relative to the current bits.

Any ideas to help this?

**Comment by skyao** [comment]
I suggest to use ONE new repository for each language translation:

1. There will be many language translations, not only Chinese I'm talking
2. Thees translations are always behind the main English version: you can't wait for all translations to be completed before you release your new version, and I also think you won't want to update all translations when you update some content.
3. So, let the community to update the content and keep up with the latest main English version 

This is not perfect, but I can't find a way good enough :).

**Comment by geeknoid** [comment]
We want to move the doc repo into the main istio/istio repo. So I suggest
we do istio/istio/istio.io/en for the main site and then we can put the
sibling sites next to each other. I'm not sure how we'd end up representing
this in the generated site. Would we want distinct domains (e.g. ch.istio.io
and fr.istio.io?) Or maybe we can compose them together like we do v-0.1
stuff we do today for old site versions.

**Comment by skyao** [comment]
If you use the same big repo, it's hard for you to control the commit permissions.

https://github.com/doczhcn/istio

This is the reps we are working on. We have a lot of contributors, and perhaps more later.

So it's hard for you to control all the language translation communities: Chinese is the first one, but not the last one.

I suggest to use different repo to save language translation content, and use sub-domain or direction to publish the content.

**Comment by ldemailly** [comment]
I think having the translations from english to be treated like community managed extensions in other repos makes sense

**Comment by skyao** [comment]
Is there any decision?

We, Chinese Translation team from China Service Mesh Community, will complete this documentation translation(except reference part which we don't make decision to translate or not) in the end of this month.

**Comment by rshriram** [comment]
Martin, I suggest we retain the main site here because the contributors are
different and typically inexperienced with the gh process that devs are
familiar with.

But one thing that I would be up for is hosting the reference docs from
istio/istio as it relates very much to the current state of code.

**Comment by geeknoid** [comment]
OK, so how's this for a proposal then:

   - istio/istio.github.io stays as it is. This is so we can have different
   procedures & ACLs for this repo relative to the main istio/istio repo.


   - istio/extensions/docs/zh is where we put Chinese docs. Other
   translations would be sibling directories. We want to use istio/extensions
   for non-core contributions so that seems like a good spot for this stuff.

Separately, we'll need to figure out how to surface multi-language docs on
the web site. Maybe just istio.io/<language name> would be the simplest
approach.

If that sounds OK, I'll go ahead and create the extensions repo. We need to
do this anyway soon in order to house adapters, so might as well get that
done.

**Comment by rshriram** [comment]
On Wed, Nov 15, 2017 at 1:25 AM Martin Taillefer <notifications@github.com>
wrote:

> OK, so how's this for a proposal then:
>
> - istio/istio.github.io stays as it is. This is so we can have different
> procedures & ACLs for this repo relative to the main istio/istio repo.
>
>
> - istio/extensions/docs/zh is where we put Chinese docs. Other
> translations would be sibling directories. We want to use istio/extensions
> for non-core contributions so that seems like a good spot for this stuff.
>
> Separately, we'll need to figure out how to surface multi-language docs on
> the web site. Maybe just istio.io/<language name> would be the simplest
> approach.
>
> If that sounds OK, I'll go ahead and create the extensions repo. We need to
> do this anyway soon in order to house adapters, so might as well get that
> done.
>

My general feeling is that there is a much lower barrier to entry for docs
than any code repo. And when people write docs, there will be a burst of
updates and then nothing. If the extensions repo is going to contain
adapters code and translated docs, then there is going to be contention
among folks there, right? Or am I missing something?

Also, do we want to take responsibility for the lost in translation mishaps
? We can’t distinguish between good and bad content in foreign language,
unless we have someone knowledgeable from a trusted party to vet the
content and hold the commit rights.


>
>
>
>
> On Mon, Nov 13, 2017 at 6:40 PM, Shriram Rajagopalan <

**Comment by geeknoid** [comment]
The point of the extensions repo is precisely to capture the fact that:

   - There are very minimal commit restrictions on this repo.


   - We invite high degree of community participation in the stuff in this
   repo.


   - We can use OWNERS files to provide controls on a per-directory basis
   for who can approve checkins.


   - We (Istio) are not generally responsible for the quality of the stuff
   in there. So the core team would not be responsible for the translated docs
   or the adapters provided by 3rd party vendors for their backends.

Basically, we're providing a convenient location for 2nd level community
contributions, making this stuff easier for people to find.

**Comment by rshriram** [comment]
Ah cool. That works then. 

Do we need some legalese declaring that we are not responsible for code/content therein?

 @Rachael-Graham any suggestions since you folks might have encountered similar situations for contributed/translated content in Bluemix docs ..

**Comment by ldemailly** [comment]
That's not exactly about translation but I think is important:
how do we do the part about docs close to the source ? (and/or generated from source/kept in sync)
(ie the reference section of the site should really be in or from istio/istio)

**Comment by geeknoid** [comment]
And more generally, the license associated with individual directories may
vary.

I'll go create the repo and include a README file that captures the gist of
my post.

**Comment by geeknoid** [comment]
The extensions repo is created and ready for business. The README.md file
will be updated shortly with a bit of context.

Laurent, regarding the main istio.github.io repo, although I was initially
interested in just making this repo a subdirectory in istio/docs, Shriram
worried me a bit in terms of making it easy to have different commit
policies and processes for docs relative to code. At this point, I'd rather
keep the repos independent.

However, I do think we want to 100% automate the production of reference
material for the site.

**Comment by Rachael-Graham** [comment]
Checking w/ Bluemix docs translation folks about any disclaimer

**Comment by Rachael-Graham** [comment]
Reply in #docs_translation: Jessica DiCarlo [12:36 PM] 
@rachael.graham In my personal opinion, I think a disclaimer for human translation isn't necessary. I've worked as a writer for IBM documentation for 10 years, and don't recall ever seeing one for human translations. But I do know that on IBM Support, machine translation is offered and does have a disclaimer. To test that, click this link, scroll to the bottom right of the page and select another language from the "Translate this page" menu. You should see a dialog that shows the disclaimer. http://www.ibm.com/support/docview.wss?uid=swg22009912 

Not sure how helpful that is to everyone, but just thought I'd let y'all know!

**Comment by rshriram** [comment]
On Thu, Nov 16, 2017 at 1:07 AM Rachael Graham <notifications@github.com>
wrote:

> Reply in #docs_translation: Jessica DiCarlo [12:36 PM]
> @rachael.graham In my personal opinion, I think a disclaimer for human
> translation isn't necessary. I've worked as a writer for IBM documentation
> for 10 years, and don't recall ever seeing one for human translations. But
> I do know that on IBM Support, machine translation is offered and does have
> a disclaimer. To test that, click this link, scroll to the bottom right of
> the page and select another language from the "Translate this page" menu.
> You should see a dialog that shows the disclaimer.
> http://www.ibm.com/support/docview.wss?uid=swg22009912
>
> Not sure how helpful that is to everyone, but just thought I'd let y'all
> know!
>

@geeknoid ^^^

**Comment by rshriram** [comment]
On Thu, Nov 16, 2017 at 7:00 AM Shriram Rajagopalan <rshriram@gmail.com>
wrote:

> On Thu, Nov 16, 2017 at 1:07 AM Rachael Graham <notifications@github.com>
> wrote:
>
>> Reply in #docs_translation: Jessica DiCarlo [12:36 PM]
>> @rachael.graham In my personal opinion, I think a disclaimer for human
>> translation isn't necessary. I've worked as a writer for IBM documentation
>> for 10 years, and don't recall ever seeing one for human translations. But
>> I do know that on IBM Support, machine translation is offered and does have
>> a disclaimer. To test that, click this link, scroll to the bottom right of
>> the page and select another language from the "Translate this page" menu.
>> You should see a dialog that shows the disclaimer.
>> http://www.ibm.com/support/docview.wss?uid=swg22009912
>>
>> Not sure how helpful that is to everyone, but just thought I'd let y'all
>> know!
>>
>
> @geeknoid ^^^
>

Rachael I think the apprehension here is about community contributed
translated content because we have no idea about the content (eg in
mandarain).

**Comment by geeknoid** [comment]
We've got a spot set up in the extensions repo to receive the docs, so I'm closing this for now.
